### PR TITLE
Some changes to the API: param names, more specialized functions

### PR DESCRIPTION
**Pull request by kgodlewski**

### CODE REVIEW

File: src/neptune_scale/__init__.py [hunk 1]
```diff
@@ -277,63 +277,89 @@ def _create_run(
...
+        self.log(step=step, timestamp=timestamp, metrics=data)
+
+    def log_fields(
+        self,
+        timestamp: Optional[datetime] = None,
+        data: Optional[Dict[str, Union[float, bool, int, str, datetime, list, set]]] = None,
+    ) -> None:
+        self.log(timestamp=timestamp, fields=data)
+
+    def log_string_sets(
```
Comment (normandy7) [inline src/neptune_scale/__init__.py:301]: Why the shift from `tags` to `string sets`? Is there a crucial difference between tag and non-tag string sets?

I think "tags" is such a familiar concept that we shouldn't hide it under more technical terms unless there's a very good reason for it.

File: src/neptune_scale/__init__.py [hunk 2]
```diff
@@ -277,63 +277,89 @@ def _create_run(
...
 
+    def log_metrics(
+        self,
+        step: Optional[Union[float, int]] = None,
+        timestamp: Optional[datetime] = None,
+        data: Optional[Dict[str, float]] = None,
+    ) -> None:
+        self.log(step=step, timestamp=timestamp, metrics=data)
+
+    def log_fields(
```
Comment (normandy7) [inline src/neptune_scale/__init__.py:294]: `fields` is still bad. Even worse given that "string set" is a type of field, too.

File: src/neptune_scale/__init__.py (hunk 1 above)
Comment (normandy7) [reply src/neptune_scale/__init__.py:301]: My suggestion:

- `log_configs()`
- `log_metrics()`
- `add_tags()`
- `remove_tags()`

If we must use a unified method for string sets, I think `manage_tags()` is better. It mirrors what you can see in the UI and is descriptive. But a downside remains: I think it's not convention to use verbs or actions as parameter names and (`log_x(add=...)` looks clunky -> I'd propose a split if possible, as in the list above.

**Review by Raalsky — APPROVED** [review]

**Comment by kgodlewski** [comment]
Note that #30 handles the `__init__`-related renaming

**Review by normandy7 — APPROVED** [review]

**Pull request merged**